ext/bcmath/bcmath.c

```diff
@@ -404,10 +404,19 @@ PHP_FUNCTION(bcpowmod)
 	php_str2num(&second, ZSTR_VAL(right));
 	php_str2num(&mod, ZSTR_VAL(modulus));
 
-	if (bc_raisemod(first, second, mod, &result, scale) != -1) {
-		RETVAL_STR(bc_num2str_ex(result, scale));
-	} else {
-		RETVAL_FALSE;
+	switch (bc_raisemod(first, second, mod, &result, scale)) {
+		case 0:
+			RETVAL_STR(bc_num2str_ex(result, scale));
+			break;
+		case -1:
+			zend_throw_exception_ex(zend_ce_division_by_zero_error, 0, "Modulo by zero");
+			break;
+		case -2:
+			//todo what kind of error should I provide here?
+			zend_argument_value_error(2, "Negative exponent not supported");
```

Comment:
This is the right exception type, but the message should be `"must be greater than 0"`.

Reply:
done